### DISCUSSION TITLE
Custom Site Settings Localization

### DISCUSSION THREAD

**Discussion by 4m-world** [OrchardCMS/OrchardCore#7127] [Q&A] [answered]
After adding the Localization part to a site setting content type, an error of "The page could not be found." is shown when you try to the localization details of the content.

**Comment by hishamco** [comment]
Please write the reproduce steps, and before that let us know which version you are working on, coz I saw such issue on one of the old builds

**Reply by 4m-world** [reply]
I'm using the latest version (dev branch)

The steps are simple I create a content type and added localization part to it. when I went to the setting screen and tried to add values to the 2nd language I pressed the language and it showed above message.

**Reply by hishamco** [reply]
As @deanmarcussen mentioned seems it's not supported yet

**Reply by 4m-world** [reply]
thank you I've seen his comment below, I will try to implement it

**Reply by hishamco** [reply]
I'm still waiting for his reply, and I will try to have a look too ...

**Comment by deanmarcussen** [comment, marked as answer]
yes, you'd be correct. 

That won't work, there is currently no support for localizing `CustomSettings`

**Reply by hishamco** [reply]
I see, is there nothing prevent to support such thing?

**Reply by deanmarcussen** [reply]
There would be nothing preventing it, but it would be a problematic piece of work.

The issue is that `CustomSettings` are handled much like a contained part, like a bag, or flowpart, taxonomy / menu with content items inside them, and the `LocalizationPart` is not designed to work with these kind of content items.

I don't know what you are trying to achieve, but if I guess correctly, and it's some strings / html content fields that need to be supplied for each culture there are two approaches I would recommend considering.

- Keep them in one `CustomSettings` type, and append the culture to the end of the field
For example `Address_En`, `Address_Fr`

- Or if the quantity of fields is to much for that, consider using a content picker on the `CustomSettings` for each culture, and create content items in the normal localizable manner, linking them for each culture with a `ContentPicker_En` or `ContentPicker_Fr` field.

I use both the above techniques where appropriate.

If it's something different you are trying to achieve please include some more details, and I may have a suggestion.

**Reply by 4m-world** [reply]
Right, I'm trying to supply branding details to a theme that I'm building, where branding details variant based on the selected culture. 
I will go with the second option as it is more flexible for my scenario.

**Reply by 4m-world** [reply]
I came with a different approach that I would like to share, I created a content type that I amended with a field named it Culture name. In this field, I enter the target culture name i.e. en, ru, ar, he

Then under the CustomSetting content type, add a bag part with the content type identified above.

In the liquid template, I run a liquid filter based on the current culture as shown below, and the result represents the desired culture output.

`{% assign detail = Site.Properties.SiteBranding.Detail.ContentItems | where: "BrandingDetail.CultureName.Text", Culture.Name | first%} `

**Reply by jptissot** [reply]
Nice workaround! I ended up creating a LocalizedTextPart that allows storing localized strings inside a single type. You can see it here: 
https://github.com/StatCan/StatCan.OrchardCore/tree/master/src/Modules/StatCan.OrchardCore.LocalizedText
https://statcan.github.io/StatCan.OrchardCore/reference/modules/LocalizedText/
